File: Models/Reservation_Model.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Shoot.Models
{
    public class Reservation_Model
    {
        [Key]
        public int Reservation_id { get; set; }

        [Required]
        public int ClientId { get; set; }

        [ForeignKey("ClientId")]
        public Client_Model? Client { get; set; }

        [Required]
        public int StadiumId { get; set; }

        [ForeignKey("StadiumId")]
        public Stadium_Model? Stadium { get; set; }

        [Required]
        [DataType(DataType.Date)]
        public DateOnly ReservationDate { get; set; }

        [Required]
        [DataType(DataType.Time)]
        public TimeOnly StartTime { get; set; }

      
        public TimeSpan Duration { get; set; } = TimeSpan.FromMinutes(90);

       
        [Required]
        public string Status { get; set; } = "Pending";
    }
}

File: ShootAI_Predictor/shoot_api/Models/reservation_model.py
from sqlalchemy import Column, Integer, String, ForeignKey, Date, Time
from sqlalchemy.ext.declarative import declarative_base

Base = declarative_base()

class Reservation(Base):
    __tablename__ = "Reservations"

    Reservation_id = Column(Integer, primary_key=True, index=True)
    ClientId = Column(Integer, ForeignKey("Client.ClientId"), nullable=False)
    StadiumId = Column(Integer, ForeignKey("Stadium.Stadium_id"), nullable=False)
    ReservationDate = Column(Date, nullable=False)

    StartTime = Column(Time, nullable=False)    # <-- FIXED (was String)
    Duration = Column(Time, nullable=False)     # <-- FIXED (was Interval)

    Status = Column(String, nullable=False, default="Pending")
